app/views/mentoring/external/show.html.haml

```diff
@@ -11,7 +11,7 @@
         Some of our best mentors worried too.
 
       - if user_signed_in?
-        - if current_user.reputation >= 20
+        - if current_user.reputation >= 20 || current_user.mentor?
```

Comment:
If they current user is a mentor they should never see this page in the first place.